File: Solutions/Solution1.java
import java.util.HashMap;
import java.util.Map;

class Solution1 {
  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> numsMap = new HashMap<>();

    for (int i = 0; i < nums.length; i++) {
      numsMap.put(nums[i], i);
    }

    for (int i = 0; i < nums.length; i++) {
      int workingTarget = target - nums[i];
      System.out.println(workingTarget);
      if (numsMap.containsKey(workingTarget) && numsMap.get(workingTarget) != i) {
        return new int[] {i, numsMap.get(workingTarget)};
      }
    }

    return new int[] {};
  }
}

File: Solutions/Solution1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dict = new Dictionary<int, int>();

        for (int index = 0; index < nums.Length; index++)
        {
            int complement = target - nums[index];

            if (dict.ContainsKey(complement))
            {
                return new int[] { dict[complement], index };
            }
            else
            {
                dict[nums[index]] = index;
            }
        }
        
        return Array.Empty<int>();
    }
}